packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -64,547 +43,759 @@ type TabsOrientationProps =
+          'iui-large': hasSublabel,
+        },
+        className,
+      )}
+      role='tablist'
+      ref={refs}
+      {...rest}
+    >
+      <TabListContext.Provider
+        value={{
+          tabsWidth,
+        }}
+      >
+        {children}
+      </TabListContext.Provider>
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div', TabListOwnProps>;
+TabList.displayName = 'Tabs.TabList';
 
-    ownerDoc.addEventListener('wheel', enableHorizontalScroll);
-  }, [overflowOptions?.useOverflow, orientation, enableHorizontalScroll]);
+// ----------------------------------------------------------------------------
+// Tabs.Tab component
 
-  const isTabHidden = (activeTab: HTMLElement, isVertical: boolean) => {
-    const ownerDoc = tablistRef.current;
-    if (ownerDoc === null) {
-      return;
-    }
+type TabOwnProps = {
+  /**
+   * Value used to associate the tab with a given panel.
+   */
+  value: string;
+  /**
+   * Tab label used for simple Tab construction.
+   * Cannot be used with tabs that have icons or descriptions.
+   */
+  label?: string | React.ReactNode;
+  /**
+   * @deprecated Don't pass `id`, as it will be automatically set.
+   */
+  id?: string;
+};
 
-    const fadeBuffer = isVertical
-      ? ownerDoc.offsetHeight * 0.05
-      : ownerDoc.offsetWidth * 0.05;
-    const visibleStart = isVertical ? ownerDoc.scrollTop : ownerDoc.scrollLeft;
-    const visibleEnd = isVertical
-      ? ownerDoc.scrollTop + ownerDoc.offsetHeight
-      : ownerDoc.scrollLeft + ownerDoc.offsetWidth;
-    const tabStart = isVertical ? activeTab.offsetTop : activeTab.offsetLeft;
-    const tabEnd = isVertical
-      ? activeTab.offsetTop + activeTab.offsetHeight
-      : activeTab.offsetLeft + activeTab.offsetWidth;
-
-    if (
-      tabStart > visibleStart + fadeBuffer &&
-      tabEnd < visibleEnd - fadeBuffer
-    ) {
-      return 0; // tab is visible
-    } else if (tabStart < visibleStart + fadeBuffer) {
-      return -1; // tab is before visible section
-    } else {
-      return 1; // tab is after visible section
-    }
-  };
+const Tab = React.forwardRef((props, forwardedRef) => {
+  const { className, children, value, label, ...rest } = props;
 
-  const easeInOutQuad = (
-    time: number,
-    beginning: number,
-    change: number,
-    duration: number,
-  ) => {
-    if ((time /= duration / 2) < 1) {
-      return (change / 2) * time * time + beginning;
+  const {
+    orientation,
+    activeValue,
+    setActiveValue,
+    type,
+    setStripeProperties,
+    idPrefix,
+    focusActivationMode,
+  } = useSafeContext(TabsContext);
+  const { tabsWidth } = useSafeContext(TabListContext);
+  const tabRef = React.useRef<HTMLButtonElement>();
+
+  const isActive = activeValue === value;
+
+  useIsomorphicLayoutEffect(() => {
+    if (isActive) {
+      tabRef.current?.scrollIntoView({
+        behavior: 'smooth',
+        block: 'center',
+        inline: 'center',
```

Comment:
i think we'll need to separate this into two different statements for vertical and horizontal, because we don't to scroll vertically in the case of horizontal tabs, and vice versa.

also should probably check for reduced-motion before using `'smooth'`

Reply:
Updated code to check motion and use different statements

Reply:
Suggest restructuring for clarity and to avoid bugs with useEffect dependencies:

```tsx
  useIsomorphicLayoutEffect(() => {
    if (isActive) {
      const behavior = isMotionOk() ? 'smooth' : 'auto';

      if (orientation === 'horizontal') {
        tabRef.current?.scrollIntoView({ behavior, inline: 'center' });
      } else {
        tabRef.current?.scrollIntoView({ behavior, block: 'center' });
      }
    }
  }, [isActive]);
```

Reply:
Updated using your second code example